modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/upgrade/v1_0_0/DeprecatedFeatureFlagUpgradeProcess.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.feature.flag.web.internal.upgrade.v1_0_0;
+
+import com.liferay.feature.flag.web.internal.company.feature.flags.CompanyFeatureFlags;
+import com.liferay.feature.flag.web.internal.company.feature.flags.CompanyFeatureFlagsProvider;
+import com.liferay.feature.flag.web.internal.model.FeatureFlag;
+import com.liferay.feature.flag.web.internal.model.FeatureFlagType;
+import com.liferay.portal.events.StartupHelperUtil;
+import com.liferay.portal.kernel.model.CompanyConstants;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.ReleaseInfo;
+
+import java.util.List;
+
+/**
+ * @author Thiago Buarque
+ */
+public class DeprecatedFeatureFlagUpgradeProcess extends UpgradeProcess {
```

Comment:
Is it necessary to create another class, or can we use the lambda function now since we've added the SF ignore rule.

Reply:
Others upgrades have this structure. Imagine we add a new upgrading regarding dev feature flags, we would not want the same function (registry) to expose two different logic (one for each upgrade). Creating a class also makes our code more organized.